testsuite/MDAnalysisTests/formats/test_libdcd.py

```diff
@@ -0,0 +1,520 @@
+
+
+@pytest.mark.parametrize("dcdfile, remarks", (
+    (DCD, DCD_HEADER), (DCD_NAMD_TRICLINIC, DCD_NAMD_TRICLINIC_HEADER),
+    (DCD_TRICLINIC, DCD_TRICLINIC_HEADER)))
+def test_header_remarks(dcdfile, remarks):
+    # confirm correct header remarks section reading
+    with DCDFile(dcdfile) as f:
+        assert len(f.header['remarks']) == len(remarks)
+
+
+@pytest.mark.parametrize("dcdfile, legacy_data, frames",
+                         ((DCD, legacy_DCD_ADK_coords, [5, 29]),
+                          (DCD_NAMD_TRICLINIC, legacy_DCD_NAMD_coords, [0]),
+                          (DCD_TRICLINIC, legacy_DCD_c36_coords, [1, 4])))
+def test_read_coord_values(dcdfile, legacy_data, frames):
+    # test the actual values of coordinates read in versus
+    # stored values read in by the legacy DCD handling framework
+    # to reduce repo storage burden, we only compare for a few
+    # randomly selected frames
+    legacy = np.load(legacy_data)
+    with DCDFile(dcdfile) as dcd:
+        for index, frame_num in enumerate(frames):
+            dcd.seek(frame_num)
+            actual_coords = dcd.read()[0]
+            desired_coords = legacy[index]
+            assert_equal(actual_coords, desired_coords)
+
+
+@pytest.mark.parametrize("dcdfile, legacy_data, frame_idx",
+                         ((DCD, legacy_DCD_ADK_coords, [5, 29]),
+                          (DCD_NAMD_TRICLINIC, legacy_DCD_NAMD_coords, [0]),
+                          (DCD_TRICLINIC, legacy_DCD_c36_coords, [1, 4])))
+def test_readframes(dcdfile, legacy_data, frame_idx):
+    legacy = np.load(legacy_data)
+    with DCDFile(dcdfile) as dcd:
+        frames = dcd.readframes()
+        xyz = frames.xyz
+        assert_equal(len(xyz), len(dcd))
+        for index, frame_num in enumerate(frame_idx):
+            assert_array_almost_equal(xyz[frame_num], legacy[index])
+
+
+def test_write_header(tmpdir):
+    # test that _write_header() can produce a very crude
+    # header for a new / empty file
+    testfile = 'test.dcd'
+    with tmpdir.as_cwd():
+        with DCDFile(testfile, 'w') as dcd:
+            dcd.write_header(
+                remarks='Crazy!',
+                natoms=22,
+                istart=12,
+                nsavc=10,
+                delta=0.02,
+                charmm=1)
+
+        with DCDFile(testfile) as dcd:
+            header = dcd.header
+            assert header['remarks'] == 'Crazy!'
+            assert header['natoms'] == 22
+            assert header['istart'] == 12
+            assert header['charmm'] == 5
```

Comment:
I don't like the fact that we say write header with `charmm=lib.DCD_IS_CHARMM` and then get back `charmm == 5` (because the current implementation forces unitcell (lib.DCD_HAS_EXTRA_BLOCKS) information.

At a minimum write the tests with the symbolic constants and comment on the fact that we are forcing it:
```python
dcd.write_header(...
   ...
   charmm=libdcd.DCD_IS_CHARMM)

...

# writer ALWAYS adds the DCD_HAS_EXTRA_BLOCKS flag
assert header['charmm'] == libdcd.DCD_IS_CHARMM | libdcd.DCD_HAS_EXTRA_BLOCKS
```